api/v1alpha1/nificluster_types.go

```diff
@@ -27,10 +27,8 @@ const (
 // NifiClusterSpec defines the desired state of NifiCluster
 type NifiClusterSpec struct {
 	// clientType defines if the operator will use basic or tls authentication to query the NiFi cluster.
-	// +kubebuilder:validation:Enum={"tls","basic"}
 	ClientType ClientConfigType `json:"clientType,omitempty"`
 	// type defines if the cluster is internal (i.e manager by the operator) or external.
-	// +kubebuilder:validation:Enum={"external","internal"}
```

Comment:
Why this change ?

Reply:
To make this kubebuilder annotation available to all the resources using the ClusterType type to avoid having to copy and paste.